deployer/src/runtime_manager.rs

```diff
@@ -15,7 +19,7 @@ type Runtimes = Arc<
         HashMap<
             Uuid,
             (
-                process::Child,
+                Arc<Mutex<process::Child>>,
```

Comment:
Do we need the `Arc<Mutex<..` here? I think `Child` is `Send` and `Sync` already [1], [2].

[1] https://docs.rs/tokio/latest/tokio/process/struct.Child.html#impl-Send-for-Child
[2] https://docs.rs/tokio/latest/tokio/process/struct.Child.html#impl-Sync-for-Child

Reply:
Yeah, but that's not the reason why I used `Mutex`.

See: https://github.com/orhun/shuttle/blob/1370efaaaf68f774094c180842d69422f0202355/deployer/src/runtime_manager.rs#L134-L140

Here, `process` is being moved to `self.runtimes` and we need to use it mutably afterwards in another closure (`tokio::spawn`) so there isn't a way to do that without an async `Mutex` and atomic reference counter since it \*does not\* implement `Clone`.

Reply:
Is this task where the process is moved the one which calls `.wait()` on it. I don't find it necessarily useful, but lmk if you disagree.

Reply:
Yeah, I tested in da38f862bcce4520f213673e51d2d451c5201f78 and it worked fine without that.

Removed in 002d0c01a408f49bfd49bbe4a82f042bdb6b5847